src/database.rs

```diff
@@ -0,0 +1,132 @@
+use std::str::FromStr;
+use std::sync::Arc;
+
+use ethers_core::types::BlockId;
+use ethers_providers::{Http, Provider};
+use pyo3::{PyErr, PyResult};
+use revm::Database;
+use revm::db::{CacheDB, DbAccount, EthersDB};
+use revm::precompile::{Address, B256};
+use revm::primitives::{AccountInfo, Bytecode, HashMap, State};
+use revm_interpreter::primitives::db::{DatabaseCommit, DatabaseRef};
+use ruint::aliases::U256;
+
+use crate::empty_db_wrapper::EmptyDBWrapper;
+use crate::utils::pyerr;
+
+type MemDB = CacheDB<EmptyDBWrapper>;
+type ForkDB = CacheDB<EthersDB<Provider<Http>>>;
+
+
+/// A wrapper around the `CacheDB` and `EthersDB` to provide a common interface
+/// without needing dynamic lifetime and generic parameters (unsupported in PyO3)
+#[derive(Clone, Debug)]
+pub(crate) enum DB {
+    Memory(Box<MemDB>),
+    Fork(Box<ForkDB>),
```

Comment:
I don't think this needs to be boxed, you can ignore the clippy warning if that's what prompted it

Reply:
I was trying to fix some borrowing problem, it was indeed not necessary. It's my first Rust PR, so thanks for the tip!

Reply:
https://github.com/gakonst/pyrevm/pull/13/commits/0dce9d83d44baa8f2e2c074f9264bad53b0c963a